haystack/components/generators/chat/openai.py

```diff
@@ -393,6 +430,7 @@ def _prepare_api_call(  # noqa: PLR0913
     ) -> dict[str, Any]:
         # update generation kwargs by merging with the generation kwargs passed to the run method
         generation_kwargs = {**self.generation_kwargs, **(generation_kwargs or {})}
+        response_format = generation_kwargs.get("response_format") if generation_kwargs else None
```

Comment:
I think that we should `pop` the response format to avoid passing it twice during the API call. Right or did I miss something?